### PR TITLE
Configable bugfix

### PR DESCRIPTION
**Pull request by vEpiphyte**
make defval's immutable
fix the getConfDefs() api

### CODE REVIEW

File: synapse/lib/config.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def addConfDef(self, name, **info):
         self._conf_defs[name] = (name, dict(info))
 
         defval = info.get('defval')
-        self._conf_opts.setdefault(name, defval)
+        self._conf_opts.setdefault(name, copy.deepcopy(defval))
```
Comment (whippit) [inline synapse/lib/config.py]: good catch :)

Comment (vertexmc) [reply synapse/lib/config.py]: :+1:

File: synapse/lib/config.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def getConfDefs(self):
         '''
         Returns the configuration definitions for this object.
         '''
-        return {name: dict(info) for (name, info) in self._conf_defs.items()}
+        return {name: dict(info[1]) for (name, info) in self._conf_defs.items()}
```
Comment (vertexmc) [inline synapse/lib/config.py]: why was this changed from `dict(info)` to `dict(info[1])`?

Comment (vertexmc) [reply synapse/lib/config.py]: makes sense

Comment (vEpiphyte) [reply synapse/lib/config.py]: The info data is a tufo form.
